src/engine/sparqlExpressions/SparqlExpression.cpp

```diff
@@ -65,6 +65,12 @@ std::unique_ptr<SparqlExpression> SparqlExpression::replaceChild(
   return std::exchange(children()[childIndex], std::move(newExpression));
 }
 
+// _____________________________________________________________________________
+std::optional<SparqlExpression*> SparqlExpression::getChildAtIndex(
+    [[maybe_unused]] size_t childIndex) const {
+  return std::nullopt;
```

Comment:
I think we have a virtual function  `children()` which can be used here. The default of  `nullopt` is really not necessary, but I think you can even make it non-virtual or `final` , because it directly relies on the correcteness of a function we already have.

Also, this function can easily be tested.